specs/EmailAddress.spec.ts

```diff
@@ -0,0 +1,419 @@
+import { describe, expect, it, test } from 'vitest';
+import { EmailAddress } from '../src';
+
+describe('Email address', () => {
+
+    it('debugs', () => {
+
+        const svo = EmailAddress.parse('"Joe Smith" info@qowaiv.org');
+    });
+
+    describe('length', () => {
+        test.each(
+            [
+                'the-total-length@of-an-entire-address.cannot-be-longer-than-two-hundred-and-fifty-four-characters.and-this-address-is-254-characters-exactly.so-it-should-be-valid.and-im-going-to-add-some-more-words-here.to-increase-the-length-blah-blah-blah-blah-bla.org',
+                'i234567890_234567890_234567890_234567890_234567890_234567890_234@long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long',
+                'Display name is ignored <i234567890(comments are ignored)_234567890_234567890_234567890_234567890_234567890_234@long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long>',
+            ])
+            ('not more then 254 (%s)', (s) => {
+
+                const svo = EmailAddress.tryParse(s);
+                expect(svo?.length).toBe(254);
+            });
+
+        it('not 255 or longer', () => {
+            const svo = EmailAddress.tryParse('i234567890_234567890_234567890_234567890_234567890_234567890_234@long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long.long1');
+            expect(svo).toBeUndefined();
+        })
+    });
+
+    describe('local part', () => {
+        test.each(Array.from(
+            "0123456789" +
+            "abcdefghijklmnopqrstuvwxyz" +
+            "ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+            "!#$%&'*+-/=?^_`{}|~"))
+            ('char %s is without limitations', (c) => {
+
+                const s = c + c + c + '@qowaiv.org';
+                const svo = EmailAddress.tryParse(s);
+                expect(svo?.toString()).toBe(s);
+            });
+
+            test.each([1, 2, 17, 64])
+            ('local with length %s', (len) => {
+
+                const s = Create.string('a', len) + '@qowaiv.org';
+                const svo = EmailAddress.tryParse(s);
+                expect(svo).toBeDefined();
+            });
+
+            test.each([1, 2, 17, 62])
+            ('quoted local with length %s', (len) => {
+
+                const s = '"' + Create.string('a', len) + '"@qowaiv.org';
+                const svo = EmailAddress.tryParse(s);
+                expect(svo).toBeDefined();
```

Comment:
Why do you only check that it's defined here and not that it is the value you expect?

Reply:
Thanks for the suggestion